lib/actions/ssh.ts

```diff
@@ -350,155 +349,95 @@ export const ssh: CommandDefinition<
+		// if we're doing a direct SSH connection locally...
+		if (
+			validateDotLocalUrl(params.applicationOrDevice) ||
+			validateIPAddress(params.applicationOrDevice)
+		) {
+			const { performLocalDeviceSSH } = await import('../utils/device/ssh');
+			return await performLocalDeviceSSH({
+				address: params.applicationOrDevice,
+				port,
+				verbose,
+				service: params.serviceName,
+			});
 		}
-		const paramChecks = maybeParamChecks!;
 
-		switch (paramChecks.target) {
-			case SSHTarget.APPLICATION:
-				exitIfNotLoggedIn();
-				// Here what we want to do is fetch all device which
-				// are part of this application, and online
-				try {
-					const devices = await sdk.models.device.getAllByApplication(
-						params.applicationOrDevice,
-						{ $filter: { is_online: true }, $select: ['device_name', 'uuid'] },
-					);
-					const choice = await selectFromList(
-						'Please choose an online device to SSH into:',
-						map(devices, ({ device_name, uuid: uuidToChoose }) => ({
-							name: `${device_name}  [${uuidToChoose.substr(0, 7)}]`,
-							uuid: uuidToChoose,
-						})),
-					);
-					// A little bit hacky, but it means we can fall
-					// through to the next handling mechanism
-					params.applicationOrDevice = choice.uuid;
-				} catch (e) {
-					if (e instanceof BalenaApplicationNotFound) {
-						exitWithExpectedError(
-							`Could not find an application named ${
-								params.applicationOrDevice
-							}`,
-						);
-					}
-					throw e;
-				}
-			case SSHTarget.DEVICE:
-				exitIfNotLoggedIn();
-				// We want to do two things here; firstly, check
-				// that the device exists and is accessible, and
-				// also convert a short uuid to a long one if
-				// necessary
-				let uuid = params.applicationOrDevice;
-				let version: string | undefined;
-				let id: number | undefined;
-				let isOnline: boolean | undefined;
-				// We also want to avoid checking for a device if we
-				// know it exists
-				if (!paramChecks.deviceChecked || validateShortUuid(uuid)) {
-					try {
-						const device = await sdk.models.device.get(uuid, {
-							$select: ['id', 'uuid', 'supervisor_version', 'is_online'],
-						});
-						uuid = device.uuid;
-						version = device.supervisor_version;
-						id = device.id;
-						isOnline = device.is_online;
-					} catch (e) {
-						if (e instanceof BalenaDeviceNotFound) {
-							exitWithExpectedError(`Could not find device: ${uuid}`);
-						}
-					}
-				} else {
-					version = paramChecks.device!.supervisor_version;
-					uuid = paramChecks.device!.uuid;
-					id = paramChecks.device!.id;
-					isOnline = paramChecks.device!.is_online;
-				}
+		// this will be a tunnelled SSH connection...
+		exitIfNotLoggedIn();
+		const uuid = await getOnlineTargetUuid(sdk, params.applicationOrDevice);
 
-				if (!isOnline) {
-					throw new Error(`Device ${uuid} is not online.`);
-				}
+		// We want to do two things here; firstly, check
+		// that the device exists and is accessible
```

Comment:
Comment went funky here

Reply:
removed it, the code is clearer now so no need to be explicit